### ISSUE TITLE
DISSAGREGATION WITH A MODEL GENERATED AND A NEW DATASET WITH ONLY THE AGREGATE(MAIN)

### ISSUE THREAD

**Issue by crn565** [nilmtk/nilmtk#976]
I think the ultimate purpose of NILMTK, given the current state of development, is to start from a model generated for a number of applications, using a new data set containing only the aggregate for the same number of applications, to obtain the disaggregated measures of the different applications.

Disaggregating a Dataset with Aggregated Data**: - Although NILMTK is primarily used to disaggregate aggregated data, I think it's possible to apply a pre-trained model to a dataset containing only the aggregated signal

Here are the general steps to do so that I've done:

-    a.  Preparing the Dataset: - Make sure our dataset contains only the aggregated signal (i.e. total consumption) 
-    b. Load the pre-trained model that you used to train with the CO algorithm - 
-    c. Applying the Model to the Aggregate Data Set: - Use the pre-trained model to disaggregate the aggregate signal into the individual appliance contributions - ¿¿¿NILMTK provides specific functions to perform this disaggregation??.
-   d.  Evaluation and Adjustment**: - Evaluate the quality of the disaggregation using standard metrics

The problem is that I can't get past section C because I get these errors: 
RuntimeError: The model needs to be instantiated before calling `disaggregate`.  The model can be instantiated by running `train`.
and changing the code this is the new error:
TypeError: object of type 'ElecMeter' has no len()

This is the code that I've written to  disagregate:

```from nilmtk import DataSet, HDFDataStore
from nilmtk.legacy.disaggregate import CombinatorialOptimisation

# Paso 1: Cargar el modelo pre-entrenado
modelo_preentrenado_path = "./data/NILM/Results/test_results.h5"
modelo = HDFDataStore(modelo_preentrenado_path)

# Paso 2: Cargar el nuevo conjunto de datos con solo la medida agregada
nuevo_dataset_path = "ualm3.h5"
nuevo_dataset = DataSet(nuevo_dataset_path) #Cargamos el archivo ualm3.h5, que contiene las medidas agregadas.
# Obtener el nombre del primer edificio y el medidor principal (mains)
building_name = list(nuevo_dataset.buildings.keys())[0]
mains_meter = nuevo_dataset.buildings[building_name].elec.mains() # se dentifica el nombre del primer edificio en el conjunto de datos y seleccionas el medidor principal.




# Paso 3: Instanciar el desagregador y cargar el modelo pre-entrenado
desagregador = CombinatorialOptimisation() #Creamos una instancia del desagregador utilizando CombinatorialOptimisation
desagregador.model = modelo  #Carga del modelo pre-entrenado en el desagregador.

# Verificar si el modelo ya está entrenado antes de llamar al método `train`
if not desagregador.model:
    # Paso 4: Entrenar el desagregador con el nuevo conjunto de datos
    desagregador.train(nuevo_dataset)  #Si el modelo no está entrenado, lo entrenamos utilizando el nuevo conjunto de datos

# Paso 5: Desagregar el nuevo conjunto de datos utilizando el modelo pre-entrenado
desagregado = desagregador.disaggregate_chunk(mains_meter)  #Utilizamos el modelo para desagregar el consumo del medidor principal.

# Guardar el resultado si es necesario
resultado_desagregado_path = "desagregado.h5"
desagregado.store(resultado_desagregado_path)
```

**Comment by crn565** [comment]
ems to be a bug in your code. The problem is in the line where I try to unbundle the mains_meter. The specific error is:
TypeError: object of type 'ElecMeter' has no len()
This error occurs because  I'me trying to get the length of an object of type ElecMeter, which is not valid.
To resolve it, I must make sure I'm using the main meter correctly. I've made sure  that mains_meter is a valid object and is set correctly before calling the disaggregate_chunk method.

To get the main meter correct, I follow these steps:

- Identify the main meter:In the  data set, I've lookes for the meter that represents the total energy consumption of the entire building or system. This is the main meter.It is usually called "mains" or "main meter" ( in my case main meter).
- Check the configuration of the main meter: I've Make sure that the main meter is correctly configured in my. I've verifyed that it is associated with the aggregated measures you wish to disaggregate.
-
